codemagic.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022 Sharezone UG (haftungsbeschränkt)
+# Licensed under the EUPL-1.2-or-later.
+#
+# You may obtain a copy of the Licence at:
+# https://joinup.ec.europa.eu/software/page/eupl
+#
+# SPDX-License-Identifier: EUPL-1.2
+
+# We are using Codemagic beside GitHub Actions for our mobile CI/CD.
+#
+# The "codemagic.yaml" allows us to configure workflows without using the
+# UI-Editor. This has the advantage that changes are tracked in Git.
+#
+# You find the documentation about the codemagic.yaml here:
+# https://docs.codemagic.io/yaml/yaml-getting-started/
+
+# With "definitions" we are able to reuse code. Therefore, we can a script once
+# and use reuse this script in other workflows without duplicating it. Just
+# define a section with &NAME and use it with *NAME.
+#
+# Docs for definitions:
+# https://docs.codemagic.io/yaml/yaml-getting-started/#reusing-sections
+#
+# Sample by Codemagic:
+# https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/yaml/yaml_anchors_aliases_sample/codemagic.yaml
+
+workflows:
+  app-preview:
+    name: App Preview
+    max_build_duration: 75
+    environment:
+      groups:
+        - github
+        - ios-publishing
+        - android_publishing
+        - certificate_credentials
+      vars:
+        BUNDLE_ID: de.codingbrain.sharezone.app
+      flutter: v2.10.5
```

Comment:
Can this somehow be read from the fvm config or `pubspec,yaml`, right?

Reply:
Technically, this should be possible, but it would take an overhead of ~1-2 minutes.

What Codemagic does:
1. Download specified Flutter version
2. Clone the repository

Therefore, we would need to this:

1. Codemagic downloads the specified (or the default one) Flutter version
2. Codemagic clones the repository
3. We read out the Flutter version specified in the `fvm_config.json`
4. Download the Flutter version
5. Update the environment variables

Reply:
Hm that kinda sucks. Do we have a document "update the flutter version" or sth where we can for now at least have the places written down where we would need to update the version string?

Reply:
Using FVM seems to be a good option 👍 The overhead is roundabout 1 min